pkg/oci/driver/driver.go

```diff
@@ -94,6 +133,33 @@ func deriveVolumeOCID(regionKey string, volumeName string) string {
 	return volumeOCID
 }
 
+// constructKubeClient uses a kubeconfig layed down by a secret via deploy.sh to return
+// a kube clientset
+func constructKubeClient() (*kubernetes.Clientset, error) {
+	fp := GetConfigDirectory() + "/kubeconfig"
+
+	c, err := clientcmd.BuildConfigFromFlags("", fp)
+	if err != nil {
+		return nil, err
+	}
+
+	k, err := kubernetes.NewForConfig(c)
+
+	return k, err
+}
+
+// lookupNodeID returns the OCID for the given nodeName.
+func lookupNodeID(k kubernetes.Interface, nodeName string) (string, error) {
+	n, err := k.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	if n.Spec.ProviderID == "" {
+		return "", errors.New("failed to find node's OCID in returned node spec")
```

Comment:
nit: this error should be something like `node is missing provider id` so it's clear where it couldn't find the ocid.